docs/migration.md

```diff
@@ -0,0 +1,193 @@
+
+## Flags
+
+The format of the Prometheus command line flags have changed.  Instead of a
+single dash, all flags now use a double dash. Common flags (`--config.file`,
+`--web.listen-address` and `--web.external-url`) are still the same but beyond
+that, almost all the storage-related flags have been removed.
+
+Some notable flags which have been removed:
+- `-alertmanager.url` In Prometheus 2.0, the command line flags for configuring
+  a static Alertmanager URL have been removed.  Alertmanager must now be
+  discovered via service discovery, see [Alertmanager service discovery](#amsd).
+
+- `-log.format` In Prometheus 2.0 logs can only be streamed to standard error.
+
+- `-query.staleness-delta` Prometheus 2.0 introduces a new mechanism for
+  handling staleness, see [Staleness and Isolation in Prometheus 2.0](https://www.youtube.com/watch?v=GcTzd2CLH7I).
+
+- `-storage.local.*` Prometheus 2.0 introduces a new storage engine, as such all
+  flags relating to the old engine have been removed.  For information on the
+  new engine, see [Storage](#storage).
+
+- `-storage.remote.*` Prometheus 2.0 has removed the already deprecated remote
+  storage flags, and will fail to start if they are supplied.  To write to
+  InfluxDB, Graphite, or OpenTSDB use the relevant storage adapter.
+
+## Alertmanager service discovery
+
+Alertmanager service discovery was introduced in Prometheus 1.4, allowing Prometheus
+to dynamically discover Alertmanager replicas using the same mechanism as scrape
+targets.  In Prometheus 2.0, the command line flags for static Alertmanager config
+have been removed, so the following command line flag:
+
+```
+./prometheus -alertmanager.url=http://alertmanager:9093/
+```
+
+Would be replaced with the following in the `prometheus.yml` config file:
+
+```yml
+alerting:
+  alertmanagers:
+  - static_configs:
+    - targets:
+      - alertmanager:9093
+```
+
+You can also use all the usual Prothetheus service discovery integrations and
+relabeling in your Alertmanager configuration.  This snippet instructs
+Prometheus to search for Kubernetes pods, in the `default` namespace, with the
+label `name: alertmanager` and with a non-empty port.
+
+```yml
+alerting:
+  alertmanagers:
+  - kubernetes_sd_configs:
+      - role: pod
+    tls_config:
+      ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+    bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+    relabel_configs:
+    - source_labels: [__meta_kubernetes_pod_label_name]
+      regex: alertmanager
+      action: keep
+    - source_labels: [__meta_kubernetes_namespace]
+      regex: default
+      action: keep
+    - source_labels: [__meta_kubernetes_pod_container_port_number]
+      regex:
+      action: drop
+```
+
+## Recording rules and alerts
+
+The format for configuring alerting and recording rules has been changed to YAML.
+An example of a recording rule and alert in the old format:
+
+```
+job:request_duration_seconds:99percentile =
```

Comment:
This would be job:request_duration_seconds:histogram_quantile99

I'd suggest a much simpler example expression.